### PR TITLE
Adding support for handling repos that don't have a gitdepend file

### PR DESCRIPTION
**Pull request by pdubb29**
issue #225

Why:

* No way to handle when config files are not found

This change addresses the need by:

* Adding an interface method to call when no configuration file is found
* Applying that interface method in all the visitors.

### CODE REVIEW

File: GitDepend/Visitors/CheckOutBranchVisitor.cs [hunk 1]
```diff
@@ -60,6 +60,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode = code;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new System.NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/CheckOutBranchVisitor.cs]: This needs to be implemented.

File: GitDepend/Visitors/CheckOutDependencyBranchVisitor.cs [hunk 2]
```diff
@@ -57,6 +57,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode.Success;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new System.NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/CheckOutDependencyBranchVisitor.cs]: This needs to be implemented.

File: GitDepend/Visitors/CreateBranchVisitor.cs [hunk 3]
```diff
@@ -64,6 +64,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode = code;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new System.NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/CreateBranchVisitor.cs]: This needs to be implemented.

File: GitDepend/Visitors/DeleteBranchVisitor.cs [hunk 4]
```diff
@@ -73,6 +73,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode = code;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new System.NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/DeleteBranchVisitor.cs]: This needs to be implemented.

File: GitDepend/Visitors/ListAllBranchesVisitor.cs [hunk 5]
```diff
@@ -65,6 +65,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode.Success;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/ListAllBranchesVisitor.cs]: This needs to be implemented.

File: GitDepend/Visitors/ListMergedBranchesVisitor.cs [hunk 6]
```diff
@@ -65,6 +65,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode.Success;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/ListMergedBranchesVisitor.cs]: This needs to be implemented.

File: GitDepend/Visitors/NullVisitor.cs [hunk 7]
```diff
@@ -36,6 +36,15 @@ public ReturnCode VisitProject(string directory, GitDependFile config)
             return ReturnCode.Success;
         }
 
+        /// <summary>
+        /// Called when the algorithm can't find the configuration file.
+        /// </summary>
+        /// <returns></returns>
+        public ReturnCode MissingConfigurationFile()
+        {
+            throw new System.NotImplementedException();
```
Comment (kjjuno) [inline GitDepend/Visitors/NullVisitor.cs]: This needs to be implemented.

**Comment by pdubb29** [comment]
Lol whoops! haha

**Pull request merged**